### DISCUSSION TITLE
How to use a dispatch parameter value and pass it to an img src or any html element attribute?

### DISCUSSION THREAD

**Discussion by melokullc** [livewire/livewire#8419] [1. Help] [answered]
Hi team,

I am hoping this will be a quick one, but I am not sure.

I am using a global notification Laravel include (this is included on the layout template view) to display notifications to users upon actions taken i.e. in this example, a user updates their name in the app, after a success save triggers, the notification appears specifying an email has been sent (to confirm the update, similar to when the user updates password or email).

So in any livewire component I can call dispatch to display the notification and pass whatever message I specify, however I wanted to also pass another value which would be a value of an icon to be used within the notification. However, this value needs to be as unique as the message depending on which livewire component is calling dispatch.

The message part of this is working perfectly, however I can't seem to find a way to use the "icon" value in the img src to specify which icon to use. Right now, I just put "icon" in the image path as a reference. I've tried using x-data, x-bind, $ref, etc... but can't seem to extract the icon value to use within img src.

Any help or guidance is welcomed,

Thank you

*****Livewire Component*****

private $icon = 'bolt.svg';
private $message = 'Email has been sent!';

$this->dispatch('success', icon: $this->icon, message: $this->message);

*****Livewire View*****

<div 

    x-data="{ open: false, message: '', icon: '' }" 
    @success.window="open = true; message=$event.detail.message; icon=$event.detail.icon"
    
    >

    <div

        x-cloak
        x-show="open"
        x-transition.out.opacity.duration.2000ms
        x-effect="if(open === true) setTimeout(() => open = false, 3000)"
        class="fixed top-5 right-5 flex w-full max-w-sm p-5 justify-evenly border-2 bg-white"

    >

        <div class="w-100">

            <img src="'/images/icons/icon" class="inline-block h-5 w-5">

            <p x-text="message" class="inline-block"></p>

        </div>
     
    </div>

</div>

*****End Result*****

![image](https://github.com/livewire/livewire/assets/63275112/790c7957-1e27-4053-a13e-f6df1749d990)

**Comment by EdgarsJoja** [comment]
If I understand correctly, this might work:
```blade
<img :src="`/images/icons/${icon}`" class="inline-block h-5 w-5">
```

Tip: use triple backticks for better multi-line code-fromatting :)

**Reply by melokullc** [reply, marked as answer]
@EdgarsJoja Thank you! I did try that out with no success unfortunately. But, you got me going on the right track again.

Eventually, I stalled out and figured I would just try to do it the way Caleb did it on the notification component with something like 

![image](https://github.com/livewire/livewire/assets/63275112/82c904a2-6569-44d6-924b-0ef802f8f836)

And just list out all the possible icon options into the notification file itself like above and just set the notification.type to the value I wanted, so that only my type/icon would trigger true and display , but that also seemed to not be super optimal with at least a few dozen options being listed.

Eventually, I ran across a tutorial of Caleb's where he used :value="framework" during his combo box headless ui component. So I switched out value for src, and this ended up working.

![image](https://github.com/livewire/livewire/assets/63275112/dcf522c3-be15-4f1d-8b4d-8235fbd2f0b4)

Thank you again!

**Comment by melokullc** [comment]
> If I understand correctly, this might work:
> 
> ```
> <img :src="`/images/icons/${icon}`" class="inline-block h-5 w-5">
> ```
> 
> Tip: use triple backticks for better multi-line code-fromatting :)
